python/ql/src/semmle/python/pointsto/PointsTo.qll

```diff
@@ -1238,7 +1238,7 @@ module Expressions {
             or
             op instanceof USub and value = ObjectInternal::fromInt(-opvalue.intValue())
             or
-            opvalue = ObjectInternal::unknown() and value = opvalue
+            not op instanceof Not and opvalue = ObjectInternal::unknown() and value = opvalue
```

Comment:
Just to double-check, this extra condition is needed because `ObjectInternal::unknown()` has a `booleanValue` (in fact, it has both such values), and therefore without this `instanceof` check, we would have that a `not` of an unknown value was both `True`, `False`, _and unknown_.

I gather `USub` doesn't present a similar problem, because `ObjectInternal::unknown()` doesn't have an `intValue`.

Reply:
Yes, exactly. We want `-unknown` and `~unknown` to be `unknown`, but `not unknown` to be `True` or `False`